lib/Pipes/FileContainer.cpp

```diff
@@ -123,6 +130,12 @@ llvm::Error FileContainer::storeToDisk(llvm::StringRef Path) const {
     return llvm::Error::success();
   }
 
+  if (Path == "-") {
+    auto Buffer = unfailableGetFileAsStream(this->Path);
+
+    llvm::outs() << Buffer->getBuffer();
+  }
```

Comment:
Can we adopt `ToolOutputFile`? It handles `-` and does not leave around stuff in case of failure.
We're already using it [elsewhere](https://github.com/revng/revng/blob/develop/include/revng/Model/ToolHelpers.h#L235).

Reply:
i could but to do so it would require to load into memory the file in the general case

```
  auto Error = errorCodeToError(llvm::sys::fs::copy_file(this->Path, Path));
  auto Perm = cantFail(errorOrToExpected(fs::getPermissions(this->Path)));
  fs::setPermissions(Path, Perm);
```

Reply:
How does that work with `-`?

Reply:
i load it and dump it to stdout

Reply:
I mean, `copy_file`, if an argument is `-` it will fail/create a file named `-`.

Reply:
yeah, that is why we have a if before everything else that check if the output path is -. In that case it executes the code in the top snippet